### PR TITLE
removed try/catch wrapping and updated tests

### PR DESCRIPTION
**Pull request by davglass**
@bengl @drewfish As discussed, this removes the `try/catch` and updates the tests to go along with it.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by bengl** [comment]
Problem: if the underlying call to `backup_object` throws, it'll go uncaught, because the `asap` call in the cache puts it into a new stack.

Might need to make the cache entirely synchronous in order to preserve the throws.

**Comment by bengl** [comment]
Looking at it, `cache.js` isn't doing anything async and never calls back with an error. 
@davglass I'd say just make it synchronous in this PR, so that we can let thrown errors bubble up. You can use `asap` in `index.js` to ensure the asynchronous-ness when calling back early due to errors.

**Comment by davglass** [comment]
@bengl From the README the cache is designed to be replaced with memcached or redis which are both async. I think that's why it was designed that way to begin with. @vsacheti thoughts?

**Comment by bengl** [comment]
@davglass after speaking with @vsacheti, it seems like as long as async cache stores are supported, anything that's thrown will need to end up being in a callback. This sucks, as it makes the dns API less transparent, but I think as long as we address it in documentation, it's sufficiently covered.

TL;DR let's keep the try-catch-passing-to-callback, but document the fact that when using dnscache, *all* errors go to the callback.

**Comment by bengl** [comment]
@davglass I'm gonna go ahead and close this in favor of #13 .
